packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -474,7 +671,22 @@ export class RoutesLoader {
             descriptor.route
           }`,
         })
-        this.app[route.method!.toLowerCase()](descriptor.route, route.handler)
+
+        /**
+         * Apply the body parser middleware if the route
+         * has not opted out of it.
+         */
+        this.applyBodyParserMiddleware(descriptor.route, route.method!)
+
+        /**
+         * If the user hasn't opted out of error handling then
+         * we wrap the handler in a try/catch block.
+         */
+        const handler = shouldWrapHandler
+          ? wrapHandler(route.handler as AsyncRouteHandler)
```

Comment:
question: This is a bit hidden from the user point of view, it could happen that the user enable the error handler but manage the wrapping himself, in that case the wrap handler here prevent that from happening since the handler does not have access to the next function. would it be better to just let the user manage that by wrapping his handler himself?

```ts
const handler = (req, res) => {}
export const GET = wrapHandler(handler)
```
something like that? would it make sense to be able to configure that behaviour?

Reply:
I think the default should be that people shouldn't need to add a wrapHandler themselves for every handler. I think it's a nice abstraction and allows people to just write actual code for their store.

But we could create a configuration to disable the default wrapHandler if a user wants to wrap everything manually, or create their own await/catch middleware, and use that instead.

```
export const config = {
  asyncHandler: false
}
```

But my question would be is there any use case for that, and would people ever want to do that? I think people would always want it to be applied if they opt-in to error handling, and if they were to create their own async handler it would end up being pretty much the same. 

If it is because people would want to extend the error handling flow, then they could add that through a custom errorHandler which makes more sense instead of doing it on the await/catch level, eg:

```
export const config = {
  errorHandler: (err, req, res, next) => {
    thirdPartyLogger.log(err) // some custom error flow

   // handle errors...
  }
}

If you see any reasons as to why people might want to do that then let me know, I might be missing something. I am very open to adding a config, if we see any reason for adding it 😄

Reply:
One reason could be when using a tool like sentry, for example, in the plugin I basically plugged myself into the errorHandler by mutating the value of the reference to it 😂 but in other scenario, they might want to use their own wrapper to send things to their tools

Reply:
But that is the case I have above where you overwrite the errorHandler middleware (which is supported in the PR), since that makes more sense rather than hooking in to the wrapHandler. That makes more sense to me at least.